### DISCUSSION TITLE
Grid / Banding from Sensor - CLAHE / Stretching

### DISCUSSION THREAD

**Discussion by StarGeezerPhil** [aaronwmorris/indi-allsky#1694] [General]
Suffering from really bad grid/banding from the sensor since setting up CLAHE and stretching for my IMX378 sensor.

I suspect part of this is sensor burnout, but it's become seriously pronounced since enabling the CLAHE and stretching (mid-tone transfer)...

...didn't know if anyone could suggest means to reduce this (or @aaronwmorris any suggestions)?

Example image:
![image](https://github.com/user-attachments/assets/74e9422e-6bee-41b4-8e49-c932737741ff)

Relevant config settings - these really brought out the definition of stars in (very) light-polluted skies, but the side effect is the banding becoming more evident. It's not too much of an issue on individual frames, but once stacked into a startrail timelapse it becomes very obvious.
![image](https://github.com/user-attachments/assets/254c8919-7fc8-42eb-8d6c-d8f53e024c67)

Single frame example below:
![image](https://github.com/user-attachments/assets/d5c829ea-8fab-42da-8fac-e93fd05b9111)

Any suggestions or tips appreciated.

**Comment by aaronwmorris** [comment]
Do you use any of the stacking/registration functions?

**Reply by StarGeezerPhil** [reply]
I disabled stacking (was previously stacking 3 at a time).

**Reply by aaronwmorris** [reply]
I doubt this effect is a problem with your sensor.  Based on your settings, I believe the grid pattern is 100% a side effect of the CLAHE algorithm.  My understanding of CLAHE is that it performs the contrast enhancement in a grid pattern (4x4 blocks in your case) and this results in a grid pattern.

I also think the grid pattern is exacerbated by the uniform/flat background of the image.   The CLAHE algorithm does better with highly dynamic images, but all sky images tend to have a relatively uniform presentation.

**Reply by StarGeezerPhil** [reply]
Thanks Aaron,

I'll disable CLAHE and see that resolves the banding/grid issue, a shame though - it was really bringing out some contrast.

Appreciated 👍
